web/src/engine/keyboard/src/lazySpan.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * A span that does not cut out the substring until it absolutely has to!
+ */
+export class LazySpan {
```

Comment:
From `common/models/wordbreakers`, with alterations (for sub-span extraction).

The original may be found at https://github.com/keymanapp/keyman/blob/72c894bf386d0bc36c48172bc0abcf77057fc338/common/models/wordbreakers/src/main/default/index.ts#L69.